hack/lib/init.sh

```diff
@@ -18,6 +18,9 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
+# Short-circuit if init.sh has already been sourced
+[[ $(type -t kube::init::loaded) == function ]] && return 0
```

Comment:
I never thought I will learn so much `bash`, @thockin used `declare -F`  in a previous PR for the `update-codegen` , and this seems to be equivalent (it seems that also more portable but we don't care since we only support bash) https://stackoverflow.com/questions/85880/determine-if-a-function-exists-in-bash

Reply:
this is saying if we already defined this function, to skip loading the rest of the file... I don't *think* that's the same

Reply:
I mean, the mechanism to detect a function exist , one uses `type -t` and other `decare -F` , it was jjust a note to myself

Reply:
re-read what `declare -F` is doing ... seems like this works as well, and seems a little more readable to me, but I don't feel strongly

Reply:
me neither, I just learn about both these days